src/test/resources/com/atomist/rug/test/gherkin/handler/event/PassingFeature1StepsAgainstGeneratedWithArrays.ts

```diff
@@ -0,0 +1,17 @@
+import {Given,When,Then, HandlerScenarioWorld} from "@atomist/rug/test/handler/Core"
+
+import {Push} from "@atomist/rug/cortex/stub/Push"
+import {Commit} from "@atomist/rug/cortex/stub/Commit"
+
+ 
+Given("a sleepy country", f => {
+})
+When("a visionary leader enters", world => {
+   world.registerHandler("ReturnsEmptyPlanEventHandlerGenWithArrays")
+   let p = new Push().addContains(new Commit())
```

Comment:
I'm adding TypeDoc on the generated `add` methods, which should make things clearer.